splunk_connector.py

```diff
@@ -1341,6 +1330,26 @@ def _run_query(self, search_query, action_result, attach_result=False, kwargs_cr
         self.debug_print('Done run query')
         return action_result.set_status(phantom.APP_SUCCESS)
 
+    def _wait_until_splunk_job_results_are_ready(self, action_result: phantom.ActionResult, job: splunk_client.Job, retry_limit: int) -> bool:
+        for attempt_count in range(0, retry_limit):
+            try:
+                #  Timing out the splunk job is required, because the job
+                #  could be stuck in permanent "QUEUED" state after the Splunk
+                #  stack has crashed.
+                max_waiting_time: float = time.time() + self.splunk_job_timeout
+                while not job.is_ready():
+                    if time.time() > max_waiting_time:
+                        return action_result.set_status(phantom.APP_ERROR, consts.SPLUNK_ERR_SPLUNK_JOB_HAS_TIMED_OUT)
+                    time.sleep(self.sleeptime_in_requests)
+                job.refresh()
+                break
+            except Exception as e:
```

Comment:
can we not generalize this Exception?
or can we debug print if exception happened?

Reply:
Adding less generalize Exception will be troublesome as we don't have any influence over what is raised by the splunk-sdk and when (but every exception should be treated the same way by us for now) so I went with general Exception. 
However, I will add debug print, which should help with tracking the loop with retries